### PR TITLE
Make docstrings consistent with numpydoc format

### PR DESCRIPTION
**Pull request by nitinkgp23**
In many places, docstrings violate the numpydoc format, hence breaking
the type hinting in PyCharm. This commit resolves all those
inconsistencies.

Fixes https://github.com/tardis-sn/tardis/issues/680

### CODE REVIEW

**Comment by nitinkgp23** [comment]
@ftsamis Can you please review the PR? :)

File: tardis/model/density.py [hunk 1]
```diff
@@ -137,19 +137,19 @@ def calculate_density_after_time(densities, time_0, time_explosion):
     scale the density from an initial time of the model to the
     time of the explosion by ^-3
 
-    Parameters:
+    Parameters
     -----------
 
-    densities: ~astropy.units.Quantity
+    densities: astropy.units.Quantity
```
Comment (ftsamis) [inline tardis/model/density.py]: Let's keep the space before the `:` consistent. There are other instances of this too.

File: tardis/montecarlo/base.py [hunk 2]
```diff
@@ -272,16 +272,16 @@ def calculate_radiationfield_properties(self):
         Parameters
         ----------
 
-        nubar_estimator : ~np.ndarray (float)
+        nubar_estimator : np.ndarray (float)
```
Comment (ftsamis) [inline tardis/montecarlo/base.py]: I [found out](https://www.jetbrains.com/help/pycharm/2016.3/type-hinting-in-pycharm.html#legacy) that PyCharm suggests `np.ndarray[float]`. Check the link for more info. Would be nice to also fix any other instances that match the table in the link

File: tardis/montecarlo/packet_source.py [hunk 3]
```diff
@@ -40,14 +40,14 @@ def create_packet_nus(self, T, no_of_packets):
...
-        no_of_packets: ~int
+        no_of_packets: int
 
         Returns
         -------
 
-            : ~numpy.ndarray
+            : numpy.ndarray
             array of frequencies
         """
```
Comment (ftsamis) [inline tardis/montecarlo/packet_source.py]: The indentation here is wrong, also no need for the leading `:` if there is no name for the return value but only a type. There are more similar instances.

File: tardis/plasma/standard_plasmas.py [hunk 4]
```diff
@@ -44,15 +44,15 @@ def assemble_plasma(config, model, atom_data=None):
...
+    config: io.config_reader.Configuration
+    model: model.Radial1DModel
+    atom_data: atomic.AtomData
         If None, an attempt will be made to read the atomic data
         from config.
 
     Returns
     -------
-    : ~plasma.BasePlasma
+    : plasma.BasePlasma
```
Comment (ftsamis) [inline tardis/plasma/standard_plasmas.py]: No need for the leading `:` as previously mentioned.

File: tardis/simulation/base.py [hunk 5]
```diff
@@ -139,7 +139,7 @@ def advance_state(self):
 
         Returns
         -------
-            converged : ~bool
+            converged : bool
```
Comment (ftsamis) [inline tardis/simulation/base.py]: Over-indentation

**Pull request closed without merging**